src/coreboot.rs

```diff
@@ -0,0 +1,180 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (C) 2020 Akira Moroo
+// Copyright (C) 2009 coresystems GmbH
+// Copyright (C) 2008 Advanced Micro Devices, Inc.
+
+use core::mem::size_of;
+
+use crate::boot::{E820Entry, Info};
+
+#[derive(Debug)]
+#[repr(C)]
+struct Header {
+    signature: [u8; 4],
+    header_bytes: u32,
+    header_checksum: u32,
+    table_bytes: u32,
+    table_checksum: u32,
+    table_entries: u32,
+}
+
+#[derive(Debug)]
+#[repr(C)]
+struct Record {
+    tag: u32,
+    size: u32,
+}
+
+impl Record {
+    pub const TAG_FORWARD: u32 = 0x11;
+    pub const TAG_MEMORY: u32 = 0x01;
+}
+
+#[derive(Debug)]
+#[repr(C)]
+struct Forward {
+    tag: u32,
+    size: u32,
+    forward: u64,
+}
+
+#[derive(Debug)]
+#[repr(C)]
+struct MemMapEntry {
+    addr: CBUInt64,
+    size: CBUInt64,
+    entry_type: u32,
+}
+
+#[derive(Debug)]
+#[repr(C)]
+struct CBUInt64 {
+    lo: u32,
+    hi: u32,
+}
```

Comment:
Wow...

Reply:
Oops, I replaced the mess with `#[repr(packed, C)]` in 94e8240